dask_cloudprovider/aws/ecs.py

```diff
@@ -904,35 +964,72 @@ async def _start(
                 )["clusters"]
             self.cluster_name = cluster_info["clusterName"]
 
-        if self._execution_role_arn is None:
-            self._execution_role_arn = (
-                self.config.get("execution_role_arn")
-                or await self._create_execution_role()
-            )
-
-        if self._task_role_policies is None:
-            self._task_role_policies = self.config.get("task_role_policies")
+        if (
+            not self._scheduler_task_definition_arn_provided
+            and not self._worker_task_definition_arn_provided
+        ):
+            if self._execution_role_arn is None:
+                self._execution_role_arn = (
+                    self.config.get("execution_role_arn")
+                    or await self._create_execution_role()
+                )
 
-        if self._task_role_arn is None:
-            self._task_role_arn = (
-                self.config.get("task_role_arn") or await self._create_task_role()
-            )
+            if self._task_role_policies is None:
+                self._task_role_policies = self.config.get("task_role_policies")
 
-        if self._cloudwatch_logs_stream_prefix is None:
-            self._cloudwatch_logs_stream_prefix = self.config.get(
-                "cloudwatch_logs_stream_prefix"
-            ).format(cluster_name=self.cluster_name)
+            if self._task_role_arn is None:
+                self._task_role_arn = (
+                    self.config.get("task_role_arn") or await self._create_task_role()
+                )
 
         if self._cloudwatch_logs_default_retention is None:
             self._cloudwatch_logs_default_retention = self.config.get(
                 "cloudwatch_logs_default_retention"
             )
 
-        if self.cloudwatch_logs_group is None:
-            self.cloudwatch_logs_group = (
-                self.config.get("cloudwatch_logs_group")
-                or await self._create_cloudwatch_logs_group()
-            )
+        if self._scheduler_logs_prefix is None or self.scheduler_logs_group is None:
```

Comment:
Suggestions are more then welcome about how to DRY this up in an idiomatic way.

Tl;dr: if the logging information is not configured, but the task definition was provided, retrieve the logging information from the task definition.